src/main/scala/se/nimsa/sbx/dicom/DicomMetaDataDAO.scala

```diff
@@ -312,22 +300,51 @@ class DicomMetaDataDAO(val driver: JdbcProfile) {
       inner join "Equipments" on "Equipments"."id" = "Series"."equipmentId"
       inner join "FrameOfReferences" on "FrameOfReferences"."id" = "Series"."frameOfReferenceId""""
     
-    var wherePart = ""
+    val query = buildMetaDataQuery(querySelectPart, startIndex, count, orderBy, orderAscending, queryProperties)
+      
+    Q.queryNA(query).list
+  }
+  
+  def querySeries(startIndex: Long, count: Long, orderBy: Option[String], orderAscending: Boolean, queryProperties: Seq[QueryProperty])(implicit session: Session): List[Series] = {
+    implicit val getResult = GetResult(r =>
+      Series(r.nextLong, r.nextLong, r.nextLong, r.nextLong, SeriesInstanceUID(r.nextString), SeriesDescription(r.nextString), SeriesDate(r.nextString), Modality(r.nextString), ProtocolName(r.nextString), BodyPartExamined(r.nextString)))
     
-    queryProperties.foreach(queryProperty => {
-      if (wherePart.length > 0) wherePart += " and "
-      wherePart += s""""${queryProperty.propertyName}" ${queryProperty.operator.toString()} '${queryProperty.propertyValue}'"""
-    })
+    val querySelectPart = """select distinct("Series"."id"),
+      "Series"."studyId",
+      "Series"."equipmentId",
+      "Series"."frameOfReferenceId",
+      "Series"."SeriesInstanceUID",
+      "Series"."SeriesDescription",
+      "Series"."SeriesDate",
+      "Series"."Modality",
+      "Series"."ProtocolName",
+      "Series"."BodyPartExamined" from "Series"
+      inner join "Studies" on "Studies"."id" = "Series"."studyId"
+      inner join "Patients" on "Patients"."id" = "Studies"."patientId"
+      inner join "Equipments" on "Equipments"."id" = "Series"."equipmentId"
+      inner join "FrameOfReferences" on "FrameOfReferences"."id" = "Series"."frameOfReferenceId""""
     
+    val query = buildMetaDataQuery(querySelectPart, startIndex, count, orderBy, orderAscending, queryProperties)
+      
+    Q.queryNA(query).list
+  }
+  
+  def buildMetaDataQuery(selectPart: String, startIndex: Long, count: Long, orderBy: Option[String], orderAscending: Boolean, queryProperties: Seq[QueryProperty]): String = {
+    var query = selectPart
+      
+    val wherePart = queryProperties.map(queryPropertyToPart(_)).mkString(" and ")
     if (wherePart.length > 0) query += s" where $wherePart"
       
     orderBy.foreach(orderByValue =>
       query += s""" order by "$orderByValue" ${if (orderAscending) "asc" else "desc"}""")
 
     query += s""" limit $count offset $startIndex"""
-      
-    Q.queryNA(query).list
+    
+    query
   }
+  
+  def queryPropertyToPart(queryProperty: QueryProperty) = 
```

Comment:
Bara liten FYI, Scala tillåter att man definierar def:s i def:s så om queryPropertyToPart BARA används av buildMetaDataQuery så kan man flytta in funktionen dit så blir den inte tillgänglig utanför.